### DISCUSSION TITLE
3.1.1

### DISCUSSION THREAD

**Discussion by sfmskywalker** [elsa-workflows/elsa-core#5169] [Design & tech]
This patch release introduces several minor improvements and bug fixes identified following the release of version 3.1, enhancing stability and functionality:

- Resolved a MongoDB exception triggered by attempting to bulk-write an empty document set: `System.ArgumentException: Must contain at least 1 request`.
- Addressed an EF Core Migrations exception: `System.MissingMethodException: Cannot dynamically create an instance of type V3_1`.
- Corrected an issue with RabbitMQ usage without MT Dispatcher, leading to a missing consumer type error.
- Solved a problem when using MassTransit without dispatcher features, resulting in a missing `IEndpointChannelFormatter` in the DI container.
- Refined code organization by moving specific ActivityExecutionContext extensions to a partial class, part of a continuous improvement process.
- Fixed a stack overflow issue within a flowchart containing the `End` activity.
- Eliminated null reference exceptions encountered when resuming or triggering workflows without an optional options object.
- Introduced a new extension method for easier access to an activity's output via strongly-typed property expressions.
- Enhanced bookmark payload functionality with support for simple strings.

**Full Changelog**: https://github.com/elsa-workflows/elsa-core/compare/3.1.0...728aa8e35b8e3755d47cfc26bfd443fec35d11b0

<hr /><em>This discussion was created from the release <a href='https://github.com/elsa-workflows/elsa-core/releases/tag/3.1.1'>3.1.1</a>.</em>

**Comment by glime-ai** [comment]
The patch release 3.1.1 for Elsa Workflows addresses several issues and introduces improvements to enhance stability and functionality. Here are the key points from the release:

- A MongoDB exception that occurred when trying to bulk-write an empty document set has been resolved.
- An EF Core Migrations exception related to dynamically creating an instance of a type has been addressed.
- An issue with RabbitMQ usage without the MassTransit (MT) Dispatcher that led to a missing consumer type error has been corrected.
- A problem when using MassTransit without dispatcher features, which resulted in a missing `IEndpointChannelFormatter` in the Dependency Injection (DI) container, has been solved.
- Code organization has been refined by moving specific ActivityExecutionContext extensions to a partial class.
- A stack overflow issue within a flowchart containing the `End` activity has been fixed.
- Null reference exceptions that were encountered when resuming or triggering workflows without an optional options object have been eliminated.
- A new extension method has been introduced for easier access to an activity's output via strongly-typed property expressions.
- Bookmark payload functionality has been enhanced with support for simple strings.

For the full changelog, you can visit the provided link to the GitHub release page: [3.1.1 Release](https://github.com/elsa-workflows/elsa-core/releases/tag/3.1.1).
